src/test/java/com/sap/oss/phosphor/fosstars/model/score/oss/ArtifactReleaseHistoryScoreTest.java

```diff
@@ -239,12 +244,141 @@ public void testFourteenMonthOldSingleVersion() {
+    ArtifactVersion version150 = new ArtifactVersion("1.5.0", now.minusDays(64));
+    ArtifactVersion version200 = new ArtifactVersion("2.0.0", now.minusDays(31));
+    ArtifactVersions artifactVersions = ArtifactVersions.of(version100, version110, version120,
+        version130, version140, version150, version200);
+
+    ArtifactReleaseHistoryScore score = new ArtifactReleaseHistoryScore();
+    Value<ArtifactVersions> versions = RELEASED_ARTIFACT_VERSIONS.value(artifactVersions);
+    Value<ArtifactVersion> version = ARTIFACT_VERSION.value(version200);
+    ScoreValue value = score.calculate(versions, version);
+    assertEquals(6.1, value.get(), DELTA);
+
+    final Optional<Value<?>> versionsValue =
+        value.usedValues().stream().filter(v -> v instanceof ArtifactVersionsValue).findFirst();
+    assertTrue(versionsValue.isPresent());
+    assertTrue(((ArtifactVersions) versionsValue.get().get()).containsAllOf(artifactVersions));
+  }
+
+  @Test
+  public void testVersionsSelectionIfNonSemanticVersionExists() {
+    LocalDateTime now = LocalDateTime.now();
+    ArtifactVersion version100 = new ArtifactVersion("1.0.0", now.minusDays(6 * 31));
+    ArtifactVersion nonSematicVersion = new ArtifactVersion("MIGHTY-1.0", now.minusDays(4 * 31));
+    ArtifactVersion version110 = new ArtifactVersion("1.1.0", now.minusDays(3 * 31));
+    ArtifactVersion version200 = new ArtifactVersion("2.0.0", now.minusDays(2 * 31));
+
+    ArtifactVersions artifactVersions =
+        ArtifactVersions.of(version100, nonSematicVersion, version110, version200);
+    ArtifactReleaseHistoryScore score = new ArtifactReleaseHistoryScore();
+    ScoreValue value = score.calculate(RELEASED_ARTIFACT_VERSIONS.value(artifactVersions),
+        ARTIFACT_VERSION.value(version110));
+    assertEquals(6.0, value.get(), DELTA);
+
+    final Optional<Value<?>> versionsValue =
+        value.usedValues().stream().filter(v -> v instanceof ArtifactVersionsValue).findFirst();
+    assertTrue(versionsValue.isPresent());
+    assertTrue(((ArtifactVersions) versionsValue.get().get()).containsAllOf(artifactVersions));
+  }
+
+  @Test
+  public void testVersionsSelectionIfArtifactIsNonSemantic() {
+    LocalDateTime now = LocalDateTime.now();
+    ArtifactVersion version100 = new ArtifactVersion("1.0.0", now.minusDays(6 * 31));
+    ArtifactVersion nonSematicVersion = new ArtifactVersion("MIGHTY-1.0", now.minusDays(4 * 31));
+    ArtifactVersion version110 = new ArtifactVersion("1.1.0", now.minusDays(3 * 31));
+    ArtifactVersion version200 = new ArtifactVersion("2.0.0", now.minusDays(2 * 31));
+
+    ArtifactVersions artifactVersions =
+        ArtifactVersions.of(version100, nonSematicVersion, version110, version200);
+    ArtifactReleaseHistoryScore score = new ArtifactReleaseHistoryScore();
+    ScoreValue value = score.calculate(RELEASED_ARTIFACT_VERSIONS.value(artifactVersions),
+        ARTIFACT_VERSION.value(nonSematicVersion));
+    assertEquals(7.3, value.get(), DELTA);
```

Comment:
Could you please add a check for `2.0.0` here? The values should be same, shouldn't they?

```suggestion
    ScoreValue valueForNotSemanticVersion = score.calculate(RELEASED_ARTIFACT_VERSIONS.value(artifactVersions),
        ARTIFACT_VERSION.value(nonSematicVersion));
    assertEquals(7.3, valueForNotSemanticVersion.get(), DELTA);
    ScoreValue valueForSemanticVersion = score.calculate(RELEASED_ARTIFACT_VERSIONS.value(artifactVersions),
        ARTIFACT_VERSION.value(nonSematicVersion));
    assertEquals(7.3, valueForSemanticVersion.get(), DELTA);
```